primary_beam_correction.py

```diff
@@ -207,63 +195,88 @@ def write_new_fits(pbc_image, path, outputFilename):
     hdr = images.header
     newhdr = hdr.copy()
     # add in required beam keywords
-    if newhdr['CLEANBMJ'] < 0:
-        # add in required beam keywords
-        newhdr['BMAJ'] = float(newhdr['HISTORY'][-2][20:30])
-        newhdr['BMIN'] = float(newhdr['HISTORY'][-2][36:48])
-        newhdr['BPA'] = float(newhdr['HISTORY'][-2][55:])
+    try:
+        if newhdr['CLEANBMJ'] < 0:
```

Comment:
This will throw the exception if `CLEANBMJ` doesn't exist and you never get the chance to check if the `BMAJ` etc. keywords are there instead.

Reply:
Tried to fix it but i would appreciate more discussion on it.

Reply:
You are checking for the beam keyword in the HISTORY if CLEANBMJ is less than 0, but what happens if CLEANBMJ isn't there at all? You should at least make the oppurtunity for trying to locate the HISTORY keywords if CLEANBMJ doesn't exist.

Reply:
I did try to address it, but i am not 100% sure if i will have addressed it fully. Thanks.